### PR TITLE
Training vimeo tab

### PR DESCRIPTION
**Pull request by tedstrauss**
Adds a new tab type to the training module, to display a Vimeo video player
that streams from our vimeo account.
Requires the addition of a CSP exception to permit streaming from here: https://player.vimeo.com/
To use, simply add a record to the certification_training table with TrainingType='vimeo' and Content= the id of your vimeo video, e.g. 7100569.

### CODE REVIEW

File: php/libraries/NDB_Client.class.inc [hunk 1]
```diff
@@ -103,6 +103,7 @@ class NDB_Client
             "Content-Security-Policy: "
             . "default-src 'self' 'unsafe-inline'; "
             . "script-src 'self' 'unsafe-inline' 'unsafe-eval'; "
+            . "frame-src https://player.vimeo.com/; "
```
Comment (tedstrauss) [inline php/libraries/NDB_Client.class.inc]: @driusan this is the line you need to sign off on. everything else is pretty straightforward.

**Comment by xlecours** [comment]
The table definition in the schema should be change I think.

**Comment by tedstrauss** [comment]
yes it probably should, good catch

**Comment by tedstrauss** [comment]
Using this feature will require the installation of the Curl PHP module. If a project does not install Curl then they can't use vimeo for training videos. But otherwise their LORIS will not be affected. So I would consider this a caveat for existing projects.

File: modules/training/ajax/getTabContent.php [hunk 2]
```diff
@@ -66,6 +66,21 @@
         $instrument . '.mp4',
         $type
     );
+} else if ($tabInformation['TrainingType'] == 'vimeo') {
+    $oembed_endpoint = 'http://vimeo.com/api/oembed';
+    $video_url       = 'http://vimeo.com/' . $tabInformation['Content'];
```
Comment (driusan) [inline modules/training/ajax/getTabContent.php]: These are http, but the cross-origin policy is https (@johnsaigle's comment from the meeting..)

File: modules/training/ajax/getTabContent.php [hunk 3]
```diff
@@ -139,4 +154,22 @@ function createTabHTML($contentType, $title, $tabVariables, $type)
...
+/**
+ * Curl helper function.
+ *
+ * @param string $url url to get
+ *
+ * @return string
+ */
+function Curl_get($url)
+{
+    $curl = curl_init($url);
```
Comment (driusan) [inline modules/training/ajax/getTabContent.php]: Curl is a PECL extension that isn't in the base PHP installation and needs to be compiled into it. I think this is a pretty heavy requirement to add just for vimeo videos.

File: modules/training/php/NDB_Form_training.class.inc [hunk 4]
```diff
@@ -190,5 +190,19 @@ class NDB_Form_Training extends NDB_Form
...
+     *
+     * @return array of javascript to be inserted
+     */
+    function getJSDependencies()
+    {
+        $factory = NDB_Factory::singleton();
+        $baseURL = $factory->settings()->getBaseURL();
+        $deps    = parent::getJSDependencies();
+        return $deps;
+    }
```
Comment (driusan) [inline modules/training/php/NDB_Form_training.class.inc]: This isn't doing anything other than returning the parent function's value. I don't think this should be overridden.

File: modules/training/ajax/getTabContent.php (hunk 3 above)
Comment (tedstrauss) [inline modules/training/ajax/getTabContent.php]: It's pretty easy to install curl. But if it is too much for projects to do, then so be it. The PR will be closed.

**Pull request closed without merging**